pkg/pillar/cmd/zedagent/handlentp.go

```diff
@@ -174,6 +174,12 @@ func createNTPSource(s *chrony.ReplySourceData,
 	if flash > 0 {
 		flash ^= chrony.NTPFlagsTests
 	}
+	// Address of the NTP peer, so destination
+	dstAddr := s.IPAddr.String()
+	// DNS name if possible
+	if dstAddrNames, err := net.LookupAddr(dstAddr); err == nil {
```

Comment:
I don't think it is a good idea to have these DNS queries inside the publish info handlers.
If the DNS service is not available or device has multiple ports and the query goes out a wrong interface, this will introduce delays of multiple seconds, slowing the entire deferred queue and delaying more important info messages, such as DeviceInfo.
Preferably, DNS reverse lookups should be done from another go routine and results could be cached until DNS response TTL runs out. Something like what we do for controller name resolution: https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/resolvercache.go
This is based on [ResolveWithPortsLambda](https://github.com/lf-edge/eve/blob/master/pkg/pillar/devicenetwork/dns.go#L101) where we also respect the association between DNS servers and device ports (so that query towards a particular DNS server does not leave via wrong interface).